File: Local API/C#.NET/Messages/Modify/Modify.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Modify
{
  class Modify
  {
    static void Main(string[] args)
    {
      AXMMCFGLib.Database objDatabase = new AXMMCFGLib.Database();
      AXMMCFGLib.Constants objConstants = new AXMMCFGLib.Constants();

      string strFilter;

      Console.WriteLine("API Module: {0}\nAPI Build: {1}\n", objDatabase.Module, objDatabase.Build);

      var objMessageDB = objDatabase.GetMessageDB();
      Console.WriteLine("Open, result: {0} ({1})", objDatabase.LastError, objDatabase.GetErrorDescription(objDatabase.LastError));
      if (objMessageDB.LastError != 0)
        return;

      strFilter = "ID>=5 And ID<=7";
      AXMMCFGLib.Message objMessage = (AXMMCFGLib.Message)objMessageDB.FindFirstMessage("", strFilter, "", 100, objConstants.FLAG_FINDFIRST_GENERICMESSAGE);
      Console.WriteLine("Filter: {0}", strFilter);
      int nCount = 0;
      while (objMessageDB.LastError == 0)
      {
        if (objMessage.TypeID == "SMS")
          ((AXMMCFGLib.SmsMessage)objMessage.GetTypeObject()).Body = "New Message Text";
        else if (objMessage.TypeID == "FILE")
          ((AXMMCFGLib.FileMessage)objMessage.GetTypeObject()).Body = "New Message Text";
        else if (objMessage.TypeID == "EMAIL")
          ((AXMMCFGLib.EmailMessage)objMessage.GetTypeObject()).BodyPlainText = "New Message Text";

        objMessageDB.Save(objMessage);
        Console.WriteLine("Save message #{0}, result: {1}", objMessage.ID, objMessageDB.LastError);
        objMessage = (AXMMCFGLib.Message)objMessageDB.FindNextMessage();
        Console.WriteLine("result: {0}", objMessageDB.LastError);

        ++nCount;
      }

      if (nCount > 0)
        objMessageDB.NotifyNonStatusUpdate(); // 'non-status-update' is a trivial update; only the UI's will update.

      Console.WriteLine("Ready.");
    }
  }
}

File: Local API/VB.NET/Messages/Modify/Modify.vb
Module Modify

  Sub Main()
    Dim objDatabase As AXMMCFGLib.Database = New AXMMCFGLib.Database()
    Dim objConstants As AXMMCFGLib.Constants = New AXMMCFGLib.Constants()
    Dim strFilter As String

    Console.WriteLine("API Module: {0}" & vbLf & "API Build: {1}" & vbLf, objDatabase.[Module], objDatabase.Build)
    Dim objMessageDB = objDatabase.GetMessageDB()

    Console.WriteLine("Open, result: {0} ({1})", objDatabase.LastError, objDatabase.GetErrorDescription(objDatabase.LastError))
    If objMessageDB.LastError <> 0 Then Return

    strFilter = "ID>=5 And ID<=7"
    Dim objMessage As AXMMCFGLib.Message = CType(objMessageDB.FindFirstMessage("", strFilter, "", 100, objConstants.FLAG_FINDFIRST_GENERICMESSAGE), AXMMCFGLib.Message)
    Console.WriteLine("Filter: {0}", strFilter)
    Dim nCount As Integer = 0

    While objMessageDB.LastError = 0

      If objMessage.TypeID = "SMS" Then
        CType(objMessage.GetTypeObject(), AXMMCFGLib.SmsMessage).Body = "New Message Text"
      ElseIf objMessage.TypeID = "FILE" Then
        CType(objMessage.GetTypeObject(), AXMMCFGLib.FileMessage).Body = "New Message Text"
      ElseIf objMessage.TypeID = "EMAIL" Then
        CType(objMessage.GetTypeObject(), AXMMCFGLib.EmailMessage).BodyPlainText = "New Message Text"
      End If

      objMessageDB.Save((objMessage)) ' Important: Use (( and )) otherwise this statement always sets LastError
      Console.WriteLine("Save message #{0}, result: {1}", objMessage.ID, objMessageDB.LastError)
      objMessage = CType(objMessageDB.FindNextMessage(), AXMMCFGLib.Message)
      Console.WriteLine("result: {0}", objMessageDB.LastError)
      nCount += 1
    End While

    If nCount > 0 Then objMessageDB.NotifyNonStatusUpdate()
    Console.WriteLine("Ready.")
  End Sub


End Module
